homeassistant/config_entries.py

```diff
@@ -2399,15 +2399,14 @@ def async_update_reload_and_abort(
         reason: str = "reauth_successful",
```

Comment:
Maybe we need a kwarg to always reload since we have some integrations that expect it to not and specific tests for it

Reply:
I'm not sure. It could also be the opposite, an optional argument to only reload if there are changes.

The rationale of the PR is to always reload to make sure the config entry is not stuck in a failed state if the auth failure is temporary. At the same time, if the auth failure is temporary, the integration should handle that *without* having the user intervene by completing the auth flow.

Reply:
I was thinking about the lowest risk option, but either way I think its fine, as there are only a few places where we specifically don't want to reload.

Reply:
Added `reload_on_entry_update` so we override it.

Reply:
I'm more confused now

Doesn't `reload_on_entry_update=False` still result in a reload if the entry is updated, but it doesn't reload if it is not updated?